linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/JSBuilders.scala

```diff
@@ -9,46 +9,35 @@
 
 package org.scalajs.linker.backend.javascript
 
-import scala.annotation.tailrec
-
-import scala.collection.mutable
+import org.scalajs.ir.Position
 
 import java.io._
-import java.util.regex.Pattern
-import java.net.{ URI, URISyntaxException }
-
-import org.scalajs.ir.Position
-import org.scalajs.io._
+import java.net.URI
 
 /** An abstract builder taking IR or JSTrees */
-trait JSTreeBuilder {
+trait JSBuilder {
   /** Add a JavaScript tree representing a statement.
    *  The tree must be a valid JavaScript tree (typically obtained by
    *  desugaring a full-fledged IR tree).
    */
   def addJSTree(tree: Trees.Tree): Unit
 
+  /** Add a chunk of JavaScript code. */
+  def addTopLevelStatement(originalLocation: URI, code: String): Unit
```

Comment:
The "TopLevel" mention is misleading, because `addJSTree` also emits a *top-level* tree, but the method name does not contain `TopLevel`. They should either both have it or not have it.